### PR TITLE
Add fx4227 [v102] Fix directory for wallpaper storage

### PR DESCRIPTION
**Pull request by electricRGB**
[JIRA Reference](https://mozilla-hub.atlassian.net/browse/FXIOS-4227)

### CODE REVIEW

File: Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift [hunk 1]
```diff
@@ -130,4 +130,28 @@ class WallpaperStorageUtility: WallpaperFilePathProtocol, Loggable {
...
         }
     }
+
+    // MARK: - Migration
+    func migrateResources(completion: (Bool) -> Void) {
+        let fileManager = FileManager.default
+        guard let documentPath = fileManager.urls(for: .documentDirectory, in: FileManager.SearchPathDomainMask.userDomainMask).first,
+              let appSupportPath = fileManager.urls(for: .applicationSupportDirectory, in: FileManager.SearchPathDomainMask.userDomainMask).first
+        else {
+            browserLog.debug("Wallpaper migration error")
```
Comment (lmarceau) [inline Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift:140]: What happens for new users, this gets called?

File: ClientTests/WallpaperMigrationUtilityTests.swift [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                                              isDirectory: &isDirectory))
+
+    }
+
+    func testMigrationKeyDoesntExist() {
+        let profile = MockProfile(databasePrefix: "wallpaperMigrationTests")
+        XCTAssertNil(profile.prefs.boolForKey(migrationKey))
+    }
+
+    func testMigrationFlow() {
```
Comment (lmarceau) [inline ClientTests/WallpaperMigrationUtilityTests.swift:54]: What happens if there's no files? Should we have a test for that?

File: Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift [hunk 3]
```diff
@@ -130,4 +130,28 @@ class WallpaperStorageUtility: WallpaperFilePathProtocol, Loggable {
...
+            return
+        }
+
+        let wallpaperDocumentDirectoryPath = documentPath.appendingPathComponent("wallpapers")
+        let wallpaperAppSupportDirectoryPath = appSupportPath.appendingPathComponent("wallpapers")
+
+        do {
+            try fileManager.moveItem(at: wallpaperDocumentDirectoryPath,
```
Comment (gutley) [inline Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift:148]: FYI - I have seen cases where sometimes a move can fail if the src and dst are not on the same container. This doesn't usually show up in the simulator, but on a real device App support directory and Documents Directory are conceptually separated. If this has already been tested on a real device then it should be OK, but you may want to add something to the catch block that tries doing a copy and remove as a fallback.

File: Shared/Prefs.swift [hunk 4]
```diff
@@ -72,6 +72,7 @@ public struct PrefsKeys {
         public static let TopSiteSection = "topSitesKey"
 
         public static let MigrationCheck = "MigrationCheck"
+        public static let WallpaperDirectoryMigrationCheck = "WallpaperDirectoryMigrationCheck"
```
[the remark this answers is not in the document]
Comment (electricRGB) [reply Shared/Prefs.swift:75]: Yeppers. Will be removing this in 103, so I tossed it in there so it's easy to remove everywhere. :)

File: Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift [hunk 5]
```diff
@@ -130,4 +130,28 @@ class WallpaperStorageUtility: WallpaperFilePathProtocol, Loggable {
...
+        let fileManager = FileManager.default
+        guard let documentPath = fileManager.urls(for: .documentDirectory, in: FileManager.SearchPathDomainMask.userDomainMask).first,
+              let appSupportPath = fileManager.urls(for: .applicationSupportDirectory, in: FileManager.SearchPathDomainMask.userDomainMask).first
+        else {
+            browserLog.debug("Wallpaper migration error")
+            return
+        }
+
+        let wallpaperDocumentDirectoryPath = documentPath.appendingPathComponent("wallpapers")
+        let wallpaperAppSupportDirectoryPath = appSupportPath.appendingPathComponent("wallpapers")
```
[the remark this answers is not in the document]
Comment (electricRGB) [reply Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift:145]: Good point.

File: Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift [hunk 6]
```diff
@@ -130,4 +130,28 @@ class WallpaperStorageUtility: WallpaperFilePathProtocol, Loggable {
...
+
+        do {
+            try fileManager.moveItem(at: wallpaperDocumentDirectoryPath,
+                             to: wallpaperAppSupportDirectoryPath)
```
[the remark this answers is not in the document]
Comment (electricRGB) [reply Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift:149]: Github.

File: Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift (hunk 1 above)
Comment (electricRGB) [reply Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift:140]: Good point. I'll modify this to be better for them. :)

File: ClientTests/WallpaperMigrationUtilityTests.swift (hunk 2 above)
Comment (electricRGB) [reply ClientTests/WallpaperMigrationUtilityTests.swift:54]: Will add. :)

Comment (electricRGB) [reply ClientTests/WallpaperMigrationUtilityTests.swift:54]: I have updated with MOAR TESTSETSETSET!

File: Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift (hunk 1 above)
Comment (electricRGB) [reply Client/Frontend/Home/Wallpapers/Utilities/WallpaperStorageUtility.swift]: Addressed. :)

**Review by lmarceau — APPROVED** [review]

**Comment by electricRGB** [comment]
testPrivatePreference_togglePBMDeletesPrivate is failing, which is an intermittently failing test.

**Pull request merged**